pkg/controller/plan/adapter/ovirt/client.go

```diff
@@ -534,15 +549,17 @@ func (r Client) removePrecopies(precopies []planapi.Precopy, vmService *ovirtsdk
 	}
 }
 
-func allJobsFinished(jobs []*ovirtsdk.Job) bool {
-	for _, job := range jobs {
-		status, _ := job.Status()
-		if status != ovirtsdk.JOBSTATUS_FINISHED && status != ovirtsdk.JOBSTATUS_FAILED {
-			return false
+func (r Client) isSnapshotRemovalFinished(events []*ovirtsdk.Event) bool {
+	for _, event := range events {
+		switch event.MustCode() {
+		case REMOVE_SNAPSHOT_FINISHED_FAILURE:
+			r.Log.Info("Snapshot removal failed!")
+			return true
```

Comment:
@bennyz pointed out that this logic is incorrect since the event might be correlated with a previous attempt.. so on a second thought, it would probably be better to switch back to checking jobs and say that:
1. if there's an ongoing job, wait for it to finish
2. otherwise, the operation succeeds if there's a job that finished successfully